crowbar_framework/app/models/node_object.rb

```diff
@@ -307,6 +307,42 @@ def default_fs=(value)
     @node["crowbar_wall"]["default_fs"] = value
   end
 
+  def disk_roles=(value)
+    @node.set["crowbar_wall"] ||= {}
+    @node.set["crowbar_wall"]["disk_roles"] = value
+  end
+
+  # remove devices that were claimed after their role was set
+  def drop_claimed_disks_from_roles
+    crowbar_wall["disk_roles"].keys.each do |name|
+      if crowbar_wall[:claimed_disks][name]
+        crowbar_wall["disk_roles"].delete(name)
+      end
+    end
+  end
+
+  def add_unclaimed_disks_to_roles
+    # Check for current unclaimed disks, and add unclaimed disk to
+    # disk_roles hash if it is not there yet.
+    # (New disk could have been added to the node)
+    unclaimed_physical_drives.each do |d|
+      unique_name = unique_device_for(d[0])
+      unless crowbar_wall["disk_roles"][unique_name]
+        crowbar_wall["disk_roles"][unique_name] = ""
+      end
+    end
+  end
+
+  def disk_roles(refresh = false)
```

Comment:
i would use an options hash and fetch from it. that gives the calling function more sense.
```ruby
def disk_roles(options = {})
  refresh = options.fetch :refresh, false
  ...
end
```